controllers/core/openstackclient_controller.go

```diff
@@ -0,0 +1,294 @@
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			corev1beta1.OpenStackClientReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+
+	_, secretHash, err := secret.GetSecret(ctx, h, instance.Spec.OpenStackConfigSecret, instance.Namespace)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			instance.Status.Conditions.Set(condition.FalseCondition(
+				corev1beta1.OpenClientReadyCondition,
+				condition.RequestedReason,
+				condition.SeverityInfo,
+				corev1beta1.OpenStackClientSecretWaitingMessage))
+			return ctrl.Result{RequeueAfter: time.Second * 10}, nil
+		}
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			corev1beta1.OpenClientReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			corev1beta1.OpenStackClientReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+
+	instance.Status.Conditions.Set(condition.FalseCondition(
+		corev1beta1.OpenClientReadyCondition,
+		condition.RequestedReason,
+		condition.SeverityInfo,
+		corev1beta1.OpenStackClientInputReady))
+
+	//
+	// create cm holding deployment script and render deployment script.
+	//
+	cmLabels := labels.GetLabels(instance, labels.GetGroupLabel("openstackclient"), map[string]string{})
+	envVars := make(map[string]env.Setter)
+
+	cms := []util.Template{
+		// ScriptsConfigMap
+		{
+			Name:               "openstackclient-sh",
+			Namespace:          instance.Namespace,
+			Type:               util.TemplateTypeScripts,
+			InstanceType:       instance.Kind,
+			AdditionalTemplate: map[string]string{},
+			Labels:             cmLabels,
+		},
+	}
+	err = configmap.EnsureConfigMaps(ctx, h, instance, cms, &envVars)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	clientLabels := map[string]string{
+		"app": "openstackclient",
+	}
+	pod := openstack.ClientPod(instance, clientLabels, configMapHash, secretHash)
+
+	op, err := controllerutil.CreateOrPatch(ctx, r.Client, pod, func() error {
+		pod.Spec.Containers[0].Image = instance.Spec.ContainerImage
+		err := controllerutil.SetControllerReference(instance, pod, r.Scheme)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+
+	if err != nil {
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			corev1beta1.OpenClientReadyCondition,
+			condition.ErrorReason,
+			condition.SeverityWarning,
+			condition.DeploymentReadyErrorMessage,
+			err.Error()))
+		return ctrl.Result{}, err
+	}
+
+	if op != controllerutil.OperationResultNone {
+		util.LogForObject(
+			h,
+			fmt.Sprintf("Pod %s successfully reconciled - operation: %s", pod.Name, string(op)),
+			instance,
+		)
+
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			corev1beta1.OpenClientReadyCondition,
+			condition.RequestedReason,
+			condition.SeverityInfo,
+			condition.DeploymentReadyRunningMessage))
+	}
+
+	instance.Status.Conditions.MarkTrue(
+		corev1beta1.OpenClientReadyCondition,
+		corev1beta1.OpenStackClientReadyMessage,
+	)
```

Comment:
I think a return is needed inside the `if op != controllerutil.OperationResultNone { }` block, otherwise lines 277-280 will still fire and immediately overwrite the `OpenClientReadyCondition` with `True`.

Reply:
thanks @abays I removed the extra Set() inside the {} and I think this should be cleaner